src/components/RefinementList/RefinementList.tsx

```diff
@@ -1,37 +1,113 @@
 /** @jsx h */
 
-import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
+import { h, createRef, Component } from 'preact';
 import cx from 'classnames';
 import { isSpecialClick, isEqual } from '../../lib/utils';
+import { PreparedTemplateProps } from '../../lib/utils/prepareTemplateProps';
 import Template from '../Template/Template';
 import RefinementListItem from './RefinementListItem';
 import SearchBox from '../SearchBox/SearchBox';
+import { RefinementListItem as TRefinementListItem } from '../../connectors/refinement-list/connectRefinementList';
+import { RefinementListOwnTemplates } from '../../widgets/refinement-list/refinement-list';
+import { RatingMenuTemplates } from '../../widgets/rating-menu/rating-menu';
+import { MenuTemplates } from '../../widgets/menu/menu';
+import { SearchBoxTemplates } from '../../widgets/search-box/types';
+import { CreateURL } from '../../types';
+
+export type RefinementListCSSClasses = {
```

Comment:
these are the same as in the refinement list widget, except marked as optional. I think they should be defined here, however I don't think it's optional

Reply:
Yes, this type is not perfect. Like `templateProps` it could be different depending of which widgets is calling the component (could be `RefinementListCSSClasses` but also `HierarchicalMenuCSSClasses`, `RatingMenuCSSClasses` or `MenuCSSClasses`).

I'll see if I can make the component generic to avoid this issue (same for `templateProps`)

Reply:
ah yes good point, they aren't exactly the same. Let's opt for repeating here then so that we can have a type with only the classes we actually want to expose in each widget (searchable doesn't exist on menu for example)

Reply:
Updated in https://github.com/algolia/instantsearch.js/commit/232974f3ac03ccb3d76852fdb56c53062854b81e, the CSSClasses type is now passed as a parameter to the generic.